src/main/java/net/pms/newgui/components/CustomJSpinner.java

```diff
@@ -0,0 +1,315 @@
+				((SpinnerIntModel) model).getStepSize()
+			));
+		}
+
+		if (enterMoveFocus) {
+			if (this.getEditor() instanceof CustomJSpinner.IntegerEditor) {
+				((CustomJSpinner.IntegerEditor) this.getEditor()).getTextField().addKeyListener(new KeyListener() {
+					@Override
+					public void keyTyped(KeyEvent e) {}
+					@Override
+					public void keyReleased(KeyEvent e) {}
+					@Override
+					public void keyPressed(KeyEvent e) {
+						if (e.getKeyCode() == KeyEvent.VK_ENTER) {
+							FocusManager.getCurrentManager().focusNextComponent();
+						}
+					}
+				});
+			}
+		}
+	}
+
+	protected JComponent createEditor(SpinnerModel model) {
+		if (model instanceof SpinnerDateModel) {
+			return new DateEditor(this);
+		} else if (model instanceof SpinnerNumberModel) {
+			return new NumberEditor(this);
+		} else if (model instanceof SpinnerListModel) {
+			return new ListEditor(this);
+		} else if (model instanceof SpinnerIntModel) {
+			return new IntegerEditor(this);
+		} else {
+			return new DefaultEditor(this);
+		}
+	}
+
+	public JToolTip createToolTip() {
+	    JToolTip tip = new HyperLinkToolTip();
+	    tip.setComponent(this);
+	    return tip;
+	}
+
+	protected class MouseWheelRoll  implements MouseWheelListener {
+
+		private int minimum, maximum, stepSize;
+		private CustomJSpinner spinner;
+
+		private MouseWheelRoll(CustomJSpinner spinner, int minimum, int maximum, int stepSize) {
+			this.spinner = spinner;
+			this.minimum = minimum;
+			this.maximum = maximum;
+			this.stepSize = stepSize;
+		}
+
+		@Override
+		public void mouseWheelMoved(MouseWheelEvent e) {
+	        if (e.getScrollType() != MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+	            return;
+	        }
+
+	        int value = (int) spinner.getValue();
+	        value -= e.getWheelRotation() * stepSize;
+	        if (value == minimum + stepSize && minimum % stepSize != 0) {
+	        	value = ((minimum / stepSize) * stepSize) + stepSize;
+	        } else if (value == maximum - stepSize && maximum % stepSize != 0) {
+	        	value = (maximum / stepSize) * stepSize;
+	        }
+			value = Math.min(Math.max(value, minimum),maximum);
+	        spinner.setValue(value);
+		}
+	}
+
+    /**
+     * An editor for a <code>JSpinner</code> whose model is a
+     * <code>SpinnerIntModel</code>.  The value of the editor is
+     * displayed with a <code>JFormattedTextField</code> whose format
+     * is defined by a <code>NumberFormatter</code> instance whose
+     * <code>minimum</code> and <code>maximum</code> properties
+     * are mapped to the <code>SpinnerIntModel</code>.
+     */
+    public static class IntegerEditor extends DefaultEditor
+    {
+        // This is here until DecimalFormat gets a constructor that
+        // takes a Locale: 4923525
+        private static String getDefaultPattern(Locale locale) {
+            // Get the pattern for the default locale.
+            ResourceBundle rb = LocaleData.getNumberFormatData(locale);
```

Comment:
@Nadahar It seems importing ```sun.util.resources.LocaleData``` breaks normal non-eclipse compiling (i.e. with plain old Oracle ```javac```).  When I first pulled this branch and compiled from the command line I got
```logtalk
[ERROR] /f/git/UniversalMediaServer/src/main/java/net/pms/newgui/components/CustomJSpinner.java:[170,33] cannot find symbol
[ERROR] symbol:   variable LocaleData
```
At the time I just commented out the problem lines and proceeded, since I was only interested in a general demo.  I had another look at it today, the explanation is [here](http://maven.40175.n5.nabble.com/Why-can-t-Maven-find-com-sun-xml-internal-messaging-saaj-util-ByteOutputStream-class-tp107361p107363.html), and the solution suggested [here](http://stackoverflow.com/a/30472582) fixes it, i.e.
```diff
diff --git a/pom.xml b/pom.xml
index 8b15c09..05ccdf8 100644
--- a/pom.xml
+++ b/pom.xml
@@ -388,6 +388,13 @@
 			<artifactId>javassist</artifactId>
 			<version>3.19.0-GA</version>
 		</dependency>
+		<dependency>
+			<groupId>dummy</groupId>
+			<artifactId>dummy</artifactId>
+			<version>1</version>
+			<scope>system</scope>
+			<systemPath>${java.home}/lib/rt.jar</systemPath>
+		</dependency>
 	</dependencies>
 	<build>
 		<defaultGoal>assembly:assembly</defaultGoal>
@@ -598,6 +605,10 @@
 					<target>1.7</target>
 					<testSource>1.7</testSource>
 					<testTarget>1.7</testTarget>
+					<compilerArgs>
+						<arg>-XDignore.symbol.file</arg>
+					</compilerArgs>
+					<fork>true</fork>
 				</configuration>
 			</plugin>
 
```
But this seems a bit hacky, and apparently ```import sun.xxx``` is more or less universally frowned on as unwise (although the eclipse people presumably don't think so, since their compiler is obviously ok with it :-).